auth/auth.go

```diff
@@ -17,22 +17,12 @@ import (
 )
 
 // basicAuth handles authentication with the houston api
-func basicAuth(username, password string) (string, error) {
+func basicAuth(username, password string, client houston.HoustonClientInterface) (string, error) {
```

Comment:
```suggestion
func basicAuth(username, password string, client houston.ClientInterface) (string, error) {
```

wdyt to rename to ClientInterface? since it's already in houston package.